### PR TITLE
Return correct result from module.run state

### PR DESCRIPTION
**Pull request by amendlik**
### What does this PR do?
Correct two bugs in the `module.run` state where the wrong result is returned.

1) When run in test mode, `module.run` should return `None`, indicating that changes would be made when not run in test mode
2) When no functions are passed to the state, it should return `False`, indicating an invocation error.

### What issues does this PR fix or reference?
None

### Previous Behavior
1) When run in test mode, `module.run` returned `True`, which indicates that no change would be made in in live mode. This is wrong because `modules.run` always executes the configured functions
2) When run with no functions passed, `module.run` returned `True`. This should return `False` to indicate an error.

### New Behavior
1) When run in test mode, `module.run` returns `None`, which indicates that changes will be made in in live mode. 
2) When run with no functions passed, `module.run` return `False`, which indicate an error.

### Merge requirements satisfied?
- [ ] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

File: salt/states/module.py [hunk 1]
```diff
@@ -1,12 +1,11 @@
-# -*- coding: utf-8 -*-
 r"""
 Execution of Salt modules from within states
 ============================================
 
 .. note::
 
     There are two styles of calling ``module.run``. **The legacy style will no
-    longer be available starting in the 3005 release.** To opt-in early to the
+    longer be available starting in the 3001 release.** To opt-in early to the
```
Comment (Ch3LL) [inline salt/states/module.py:8]: this should remain 3005

**Review by Ch3LL — CHANGES_REQUESTED** [review]
could you also add a changelog entry as documented here: https://docs.saltstack.com/en/master/topics/development/changelog.html

File: salt/states/module.py (hunk 1 above)
Comment (amendlik) [reply salt/states/module.py:9]: Fixed.

Comment (Ch3LL) [reply salt/states/module.py:9]: I'm not seeing this resolved or the changelog entry. Also looks like pre-commit is failing now.

Comment (amendlik) [reply salt/states/module.py]: You're right - I didn't push the changes properly. It should be fixed now

**Review by Ch3LL — DISMISSED** [review]

**Review by krionbsd — DISMISSED** [review]

**Comment by Ch3LL** [comment]
@amendlik looks like there are some merge conflicts that need to be cleaned up.

**Comment by amendlik** [comment]
@Ch3LL Ok, I rewrote my changes so they will merge with the updated files. What else do I need to do to get this merged?

**Comment by amendlik** [comment]
@Ch3LL I would like to get this merged. What steps remain?

**Comment by amendlik** [comment]
@Ch3LL I would like to get this merged. What steps remain?

**Comment by amendlik** [comment]
@Ch3LL I would like to get this merged. What steps remain?

**Review by krionbsd — DISMISSED** [review]

**Comment by sagetherage** [comment]
re-running the failed windows test to see if it is flakey or not, will check back later today on results

**Review by Ch3LL — DISMISSED** [review]

File: tests/unit/states/test_module.py [hunk 2]
```diff
@@ -128,6 +128,34 @@ def test_run_module_not_available(self):
...
+
+    def test_run_module_noop(self):
+        """
+        Tests the return of module.run state when no module function is provided
+        :return:
+        """
+        with patch.dict(module.__salt__, {}, clear=True), patch.dict(
+            module.__opts__, {"test": True, "use_superseded": ["module.run"]}
+        ):
+            ret = module.run(**{})
```
Comment (s0undt3ch) [inline tests/unit/states/test_module.py:155]:
```suggestion
            ret = module.run()
```
Is `**{}` really necessary?

Comment (amendlik) [reply tests/unit/states/test_module.py]: Good catch. I have removed that.

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**